plutus-book/doc/09-extended.adoc

```diff
@@ -62,83 +66,133 @@ sections of a Haskell program, called Plutus Tx, which are then compiled into
 Plutus Core.
 
 Scripts are a way to add smart contract functionality
-to the extended UTxO model.
+to the extended UTXO model. The code inside Plutus scripts defines the
+contract functionality.
 Many of the subsequent chapters of this book discuss building scripts
 that define specific smart contracts. In this section, however,
-we discuss in general how scripts are used in the extended UTxO model.
-For a draft of the formal specification of the extended UTxO model with
+we discuss in general how the EUTXO model enforces the functionality defined
+by the contracts in the scripts.
+For a draft of the formal specification of the extended UTXO model with
 scripts, see <<endmatter#scripts>>.
 
-Scripts, like witnesses, are another
-tool to make sure a transaction is performing only authorized actions.
-The purpose of witnessing for (non-script) spending of outputs is to verify
+The purpose of witnessing is to verify
 that the wallet spending the output has the private key associated with the address
-in the corresponding UTxO entry (i.e. the address from which it is spending the
+in the corresponding UTXO entry (i.e. the address from which it is spending the
 money) by checking that it was indeed signed (witnessed)
 by that key.
+Scripts, like witnesses, are another
+tool to ensure a transaction is spending only outputs it is authorized to.
```

Comment:
I think you're conflating witnesses and signatures.

I think *witnesses* are a piece of data attached to a transaction that shows that the transaction is authorized to spend from a particular address, with *signatures* then being the witnesses for public key addresses, and *validators* (plus the redeemer on the input) being the witnesses for script addresses.

Might want to check the ledger spec again to see how they use it.